src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -29,11 +45,16 @@ export const TodoFilter: React.FC = () => {
 
         <span className="icon is-right" style={{ pointerEvents: 'all' }}>
           {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button
-            data-cy="clearSearchButton"
-            type="button"
-            className="delete"
-          />
+          {query && (
+            <button
+              data-cy="clearSearchButton"
+              type="button"
+              className="delete"
+              aria-label="clear search"
+              onClick={() => dispatch(filterActions.setQuery(''))}
```

Comment:
The onClick function can be extracted to a separate function for better readability and maintainability.